### ISSUE TITLE
Minor docstring problems in inotify.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4750]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#4750|
|Type|defect|
|Created|2010-12-01 02:30:58Z|
|Branch|https://github.com/twisted/twisted/tree/inotify-doc-fix-4750|

1. 

The use of `self` in the module docstring here is confusing:

```
    def notify(self, filepath, mask):
        print "event %s on %s" % (
            ', '.join(inotify.humanReadableMask(mask)), filepath)
```

`self` in this case is a `_Watch` object.

2.

`_isWatched` returns a `bool`, not a watchdescriptor.

```
    def _isWatched(self, path):
        """
        Helper function that checks if the path is already monitored
        and returns its watchdescriptor if so.
```

Attachments:

* [issue_4750.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/200/20035669ad7aa73c1c0a0b590bda0aed031de80e/26f3ed972d9bb65d5b877769701ab31bc1a20ecb.patch) (1175 bytes) - added by psykidellic on 2011-02-03 20:47:19Z - Patch for docstring

<details><summary>Searchable metadata</summary>

```
trac-id__4750 4750
type__defect defect
reporter__ivank ivank
priority__low low
milestone__ 
branch__branches_inotify_doc_fix_4750 branches/inotify-doc-fix-4750
branch_author__psykidellic psykidellic
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1291170658000000 1291170658000000
changetime__1296796765000000 1296796765000000
version__None None
owner__ 
cc__psykidellic
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="psykidellic's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|psykidellic commented|
|-|-|

Hmm, I am trying to create a patch for this. I thought, I would change it to:

notify(watch, filepath, mask)

but, inotify._Watch seems to be private object. So would it make sense to expose it to a public callable? Or should _Watch become Watch?

**Comment by twisted-trac** [comment]
|<img alt="psykidellic's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|psykidellic commented|
|-|-|

Okay, as for second point - in another part of the code specifically: inotify.watch, we have:

<code>
            wd = self._isWatched(path)
            if wd:
                return wd
</code>

in which case its being used as a watch descriptor.

Also, for the watch method docstrings, I dont see anything that says it returns a watchdescriptor but in the code we have:
            

return self._addWatch(path, mask, autoAdd, callbacks)


which actually returns a watchdescriptor.

**Comment by twisted-trac** [comment]
|<img alt="psykidellic's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|psykidellic commented|
|-|-|

On closer look, fixing issue 4708 will make point 2 invalid. To correctly fix 4708, we need to return the watchdescriptor or None otherwise.

**Comment by twisted-trac** [comment]
|<img alt="psykidellic's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|psykidellic commented|
|-|-|

Find attached my patch to the docstring for point 1. Point 2, is invalid if we fix issue issue 4708.

As for point 1, after discussing with exarkun at IRC, seems the first parameter is useless and should be ignored. All that is required is correctly passed as 2nd and 3rd parameter to the callback.


So its probably better just to let user know about that.

The above patch applies to rev 30608 correctly.

**Comment by twisted-trac** [comment]
|<img alt="psykidellic's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|psykidellic commented|
|-|-|

Forgot to add the review keyword.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph removed owner|
|-|-|

Thanks for the patch.

Please un-assign review tickets so they can be reviewed by the first available person, and you can get a faster review.  My specific expertise is not required here. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [30610]) Branching to 'inotify-doc-fix-4750'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [30611]) Apply issue_4750.patch 

refs [#4750](https://github.com/twisted/twisted/issues/4750)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @exarkun|
|-|-|

Thank you very much for the patch!

There are just a couple minor points about the patch:

  1. Copyright date on modified files should be updated to include the current year
  1. I like the _ignored_ approach on the example callback, but I think the `_` is a little too much.  It's a local in the application programmer's code, so it's hard to claim it's a private Twisted implementation detail. ;)  I think _ignored_ and the docs will get the point across sufficiently well.

Otherwise great.  I'll address these two trivial points and then apply the change to trunk.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [30614]) Merge inotify-doc-fix-4750

Author: psykidellic
Reviewer: exarkun
Fixes: [#4750](https://github.com/twisted/twisted/issues/4750)

Change the documentation for the first parameter of the inotify callback interface
to indicate that it is not a very useful parameter and it should be ignored.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
